src/main/java/org/folio/innreach/batch/contribution/service/ContributionJobRunner.java

```diff
@@ -248,45 +253,81 @@ public void runItemContribution(UUID centralServerId, Instance instance, Item it
     });
   }
 
-  public void runItemMove(UUID centralServerId, Instance newInstance, Instance oldInstance, Item item) {
-    log.info("Validating item {} for moving to a new instance id : {} from old instance id : {} on central server {}", item.getId(), newInstance.getId(), oldInstance.getId(), centralServerId);
-
-    boolean eligibleItem = isEligibleForContribution(centralServerId, item);
-    boolean contributedItem = isContributed(centralServerId, oldInstance, item);
-    log.info("eligibleItem: {}, contributedItem: {}", eligibleItem, contributedItem);
-    if (!eligibleItem && !contributedItem) {
-      log.info("Skipping ineligible and non-contributed item id: {}, new instance id: {}, old instance id: {}", item.getId(), newInstance.getId(), oldInstance.getId());
-      return;
+  public void runItemContribution(UUID centralServerId, Instance instance, Item item, OngoingContributionStatus ongoingContributionStatus) {
+    try {
+      log.info("Ongoing: validating item {} for contribution to central server {} with instance id: {}", item.getId(), centralServerId, instance.getId());
+      boolean eligibleItem = isEligibleForContribution(centralServerId, item);
+      boolean contributedItem = isContributed(centralServerId, instance, item);
+      log.info("Ongoing: eligibleItem: {}, contributedItem: {}", eligibleItem, contributedItem);
+      if (!eligibleItem && !contributedItem) {
+        log.info("Ongoing: skipping ineligible and non-contributed centralServer id: {}, item id: {}, instance id : {}", centralServerId, item.getId(), instance.getId());
+        ongoingContributionStatusService.updateOngoingContribution(ongoingContributionStatus, SKIPPING_INELIGIBLE_MSG, FAILED);
+        return;
+      }
+      if (isEligibleForContribution(centralServerId, instance)) {
+        log.info("Ongoing: Re-contributing instance to update bib status, centralServer id: {}, instance id : {}, item id: {}", centralServerId, instance.getId(), item.getId());
+        recordContributionService.contributeInstance(centralServerId, instance);
+        if (eligibleItem) {
```

Comment:
Not sure if this if condition is again needed ? I think the outside if is probably doing the same thing ?